torchvision/models/inception.py

```diff
@@ -123,7 +125,9 @@ def __init__(
                     import scipy.stats as stats
                     stddev = m.stddev if hasattr(m, 'stddev') else 0.1
                     X = stats.truncnorm(-2, 2, scale=stddev)
-                    values = torch.as_tensor(X.rvs(m.weight.numel()), dtype=m.weight.dtype)
+                    values = torch.as_tensor(
+                        X.rvs(m.weight.numel(), random_state=random_state_numpy),
```

Comment:
oh, during our convo I had missed that the `rvs` was within the model's code, I thought it was just in the test. That's a bigger problem than I thought

@fmassa, googlenet and inception call scipy's rvs methods, which draw from numpy's RNG.

That's fairly unexpected I think. Shouldn't we just be relying on pytorch's RNG? I'm thinking of the following workarounds:

- add a new np_random_state parameter to the constructor to control that RNG
- rely on torch instead of numpy to draw samples from a truncated normal.

I think the second would make much more sense, although I don't know how easy this will be. WDYT?


As a temporary workaround we could use a pytest fixture that sets numpy's RNG and restores it as e.g. in https://gist.github.com/VictorDarvariu/6cede9c79900c6215b5f848993d283c6, but ugh

Reply:
> That's fairly unexpected I think. Shouldn't we just be relying on pytorch's RNG? I'm thinking of the following workarounds:

We should, but PyTorch didn't implement `trunc_normal` back when we first implemented this model.
It seems now that it's implemented in https://github.com/pytorch/pytorch/pull/32397 , so we could replace it to use PyTorch's implementation. It should be fairly straightforward, but it would be good to check if the PyTorch sampling is much slower than scipy's or not, and to make this change in a separate PR.

Reply:
@vmoens Would you like a submit a PR to change googlenet and inception to rely on `torch.nn.init.trunc_normal_` instead?

I think we can keep this one on hold until then

Reply:
Sure let me do this: 
- keep `np.random.seed(0)` in set_seed for now in this PR
- do a new PR where `np.random.seed(0)` is taken away and we use `trunc_normal_` for inception

Reply:
To keep matters separate, we should aim at removing all numpy seedings in one go, so the first step might unnecessary. We'll need to get rid of the `rvs` calls first